### ISSUE TITLE
GDScript: serious HotReload issue regarding Lambdas 

### ISSUE THREAD

**Issue by Ryan-000** [godotengine/godot#88756]
### Tested versions

4.2, 4.2.1.  I tried 4.1.3 but it seems to sometimes suffer from another issue, and in a different way. 

### System information

Godot v4.2.1.stable.mono, Desktop

### Issue description

I got [this plugin](https://github.com/godot-extended-libraries/godot-debug-menu) a while back, and all was good (or so I thought).

Forward to now, I'm using GDScript more, and I'm wondering why are all my lambda functions just “randomly” not working?

Turns out there was something in the plugin causing an engine problem.

### Steps to reproduce

Turns out adding a script like this, and having it instanced at some point in runtime in your scene or with .new().
```gdscript
extends Node

# comment and uncomment this, to test. (do it before running the project)
var hi = func literally_some_function(): pass
```
Will completely ruin your hot reload as ALL lambdas, in every single GDScript suddenly break when you do it. 
(And you don't get any error that something went wrong 😱)

Below is an example, you can see the point where the script breaks (that's when I save to hot reload)
![image](https://github.com/godotengine/godot/assets/73148864/838a77d7-327a-4da0-a018-cdc926d5a089)

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/godotengine/godot/files/14392502/mrp.zip)

**Comment by dalexeev** [comment]
CC @rune-scape @vnen @RandomShaper

**Comment by rune-scape** [comment]
fixed by https://github.com/godotengine/godot/pull/86860 (just tested the mrp also)
the fix will be in the 4.2.2 release

this also happens when using [PankuConsole](https://github.com/Ark2000/PankuConsole), the original reason why i made the lambda hot reload PR
its made worse there by every error being printed using a lambda which creates a never ending loop of error if u hot reload once,, it slowed my game development so much

**Comment by dalexeev** [comment]
Closing according to the comment above. Thanks for the fix and testing, @rune-scape!

@Ryan-000 Please try 4.2.2 RC 1 and/or 4.3 dev 3. Thanks for the report and let me know if your issue is fixed.
